src/lib/Parser.hs

```diff
@@ -260,6 +262,56 @@ topLet = do
         k = if v == mkName "eff" then EffectRowKind else TypeKind
         uTyKind = WithSrc (Just pos) $ UPrimExpr $ TCExpr k
 
+interfaceDef :: Parser [UDecl]
+interfaceDef = do
+  keyWord InterfaceKW
+  -- TODO: allow multi-parameter type classes
+  interfaceName <- upperName <|> symName
+  (oneAnnBinder, pos) <- withPos annBinder
+  keyWord WhereKW
+  record <- withSrc $ URecordTy <$> interfaceRecordFields ":"
+  let tyCon = UConDef interfaceName $ toNest [oneAnnBinder]
+  let consName = mkInterfaceConsName interfaceName
+  let varTypeName = case oneAnnBinder of
+        Bind v   -> varName v
+        -- This will fail during typechecking.
```

Comment:
Then it should be a TODO, right? Or is that expected? If yes, please explain why!

Reply:
I intended the message to mean "we're good, this will fail as intended".
The only case when that second case is reached is when the type variable is given "`_`" for a name, as `annBinder` allows this; I don't believe that this should succeed, but maybe this could be an error at the parser level, WDYT?

Reply:
Yeah, I mean. If you know that something is going to fail then it's better to give an informative error message

Reply:
This is resolved by instead allowing unnamed parameter types as discussed offline (see `a2f415a`)